pkg/keys/ecdsa.go

```diff
@@ -59,9 +61,16 @@ func (p *p256Verifier) MarshalPublicKey() *data.PublicKey {
 }
 
 func (p *p256Verifier) UnmarshalPublicKey(key *data.PublicKey) error {
-	if err := json.Unmarshal(key.Value, p); err != nil {
+	// Prepare decoder limited to 512Kb
+	dec := json.NewDecoder(io.LimitReader(bytes.NewReader(key.Value), 512*1024))
```

Comment:
The `key.Value` is a byte slice, so if it's large, it's already in memory and has potentially already caused an OOM. What protection does this limit reader add?

Reply:
You're right.